modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -295,6 +295,17 @@ DataLayout getDataLayout(
     return it != data_layouts.end() ? it->second : DATA_LAYOUT_UNKNOWN;
 }
 
+static
+bool hasAllOnes(const Mat &inputs, int startPos, int endPos)
+{
+    for (int i = startPos; i < endPos; i++)
+    {
+        if (inputs.at<int>(0, i) != 1 || inputs.at<int>(0, i) != -1)
```

Comment:
> `inputs.at<int>(0, i)`

Consider using single arg version instead: `inputs.at<int>(i)`

---

It makes sense to add extra checks before using of low-level processing:
```
CV_CheckLE(inputs.dims, 2, "");
CV_CheckGE(startPos, 0, "");
CV_CheckLE(startPos, endPos, "");
CV_CheckLT((size_t)endPos, inputs.total(), "");
```

Reply:
@alalek, thank you for the review!